Makefile

```diff
@@ -1,9 +1,11 @@
 .PHONY: test test-integration build run proto-api proto-lint clean download install-tools generate
 
 VERSION=`./scripts/get-tag.sh`
+GO_VERSION_CHECK=`./scripts/check-go-version.sh`
 
 # The build target should stay at the top since we want it to be the default target.
 build: pkg/web/ui/dist
+	@echo ${GO_VERSION_CHECK}
```

Comment:
Reporting the version is great, but it would be good if we could fail the build.

Reply:
should we? conduit can be built using go1.17 now because we're not using generics right?

Reply:
That's a good point. In that case, though, we shouldn't say that 1.18 is the minimum required version, if things work with 1.17?

Reply:
That sounds good to me. :+1:

Reply:
Wait a sec, even though it can be built with 1.17 right now that might break the first time we use a 1.18 feature. Our local versions are all 1.18 and our CI jobs use 1.18, we can only "guarantee" the build will work if you use 1.18, so I would side towards being strict and enforce 1.18.

Reply:
Personally, I don't mind the strictness if it's generally easy to switch between Go versions (I always use only one version). Otherwise, given that 1.18 is pretty new, many projects are still likely not using, and so likely many developers are not using it either.

Reply:
I'd have the same opinion if we were writing a library and the main use of the code would be for other people to use it as a dependency. But since we are writing a tool where the main use case is for a person to download a pre-built binary or the Docker image I see little harm in using the latest version that might not be widely adopted yet, at least from the end-user's point of view. And again, since we can only guarantee that Conduit works as expected if the developer uses the expected tooling, I think we should be strict about it.

Reply:
I'm also thinking that the dev community (i.e. our contributors) could have troubles with it, but on one hand there aren't many right now, and together with the point about the main use case (prebuilt binary), being strict here is fine.